apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java

```diff
@@ -190,6 +195,23 @@ public void setImmediateRecycling(boolean enable) {
     public void start() {
     }
 
+    @Override
+    public synchronized void reportPartialTransaction(Transaction transaction) {
+        if (closed) {
+            return;
+        }
+        if (!enabledImmediateRecycling) {
+            partialTransactions.add(transaction);
+        }
+        if (partialTransactionHandler != null) {
+            partialTransactionHandler.accept(transaction);
+        }
+    }
+
+    public void setPartialTransactionHandler(Consumer<Transaction> transactionHandler) {
+        this.partialTransactionHandler = transactionHandler;
+    }
```

Comment:
[minor] seems unused.

Reply:
It is actually used by the `PartialTransactionTest` to plugin a real `PartialTransactionReporter` into the `MockReporter`.